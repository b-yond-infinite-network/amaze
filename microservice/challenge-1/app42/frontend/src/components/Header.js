import React, {Component} from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

export default class Header extends Component {

  logoStyle = {
    width: "64px"
  };
  rowStyle = {
    backgroundColor: "#ecf0f1"
  };

  render() {
    return (
      <Row>
        <Col md={12} className="text-left" style={this.rowStyle}>
          <img style={this.logoStyle}
               src="http://pluspng.com/img-png/png-user-icon-multy-user-icons-512.png"
               alt=""/>
        </Col>
      </Row>
    )
  }

}
