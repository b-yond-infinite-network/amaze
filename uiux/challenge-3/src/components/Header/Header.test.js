import Header from './Header';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from '../../store'

const initialState = {
};

describe('Header Component', () => {

    let mockStore;

    let tree;
    let wrapper;
    let render;

    describe('has currentUser', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)(initialState);

            const props = {
                appName: 'test appName',
                currentUser: {
                    username: 'Test user',
                    image: ''
                }
            };

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <Header {...props} />
                </MemoryRouter>
            </Provider>;
            wrapper = mount(tree);
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should NOT render the login link', () => {
            const component = wrapper.find('#header-login');
            expect(component.length).toBe(0);
        });

    });

    describe('does NOT hav currentUser', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)(initialState);

            const props = {
                appName: 'test appName',
                currentUser: null
            };

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <Header {...props} />
                </MemoryRouter>
            </Provider>;
            wrapper = mount(tree);
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should render the login link', () => {
            const component = wrapper.find('#header-login');
            expect(component.length).toBe(1);
        });

    });

});