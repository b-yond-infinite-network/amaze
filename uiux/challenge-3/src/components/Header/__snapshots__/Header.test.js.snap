// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component does NOT hav currentUser Should match the snapshot 1`] = `
<nav
  className="navbar navbar-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      test appname
    </a>
    <ul
      className="nav navbar-nav pull-xs-right"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
        id="header-login"
      >
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </li>
      <li
        className="nav-item"
        id="header-register"
      >
        <a
          className="nav-link"
          href="/register"
          onClick={[Function]}
        >
          Sign up
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Header Component has currentUser Should match the snapshot 1`] = `
<nav
  className="navbar navbar-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      test appname
    </a>
    <ul
      className="nav navbar-nav pull-xs-right"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/editor"
          onClick={[Function]}
        >
          <i
            className="ion-compose"
          />
           New Post
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/settings"
          onClick={[Function]}
        >
          <i
            className="ion-gear-a"
          />
           Settings
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/@Test user"
          onClick={[Function]}
        >
          <img
            alt="Test user"
            className="user-pic"
            src=""
          />
          Test user
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
